src/tools/pdb.py

```diff
@@ -177,15 +185,18 @@ def psfwrite(filename, system, maxdist=None, molsize=4, typenames=None):
       ypos   = particle.pos[1]
       zpos   = particle.pos[2]
       type   = particle.type
+      q      = particle.q
+      mass   = particle.mass
       if typenames:
 	  name=typenames[type]
-      st = "%8d T%03d %4d UNX  %2s   %2s                    \n" % (pid+addToPid, type, mol, name, name)
+      st = "%8d T%5d    UNX  %-4s %-4s  %9.6f%14.4f\n" % (pid+addToPid, mol, name, name, q, mass)
```

Comment:
I got very confused by looking at this line not realizing that it is about the psf format, I guess the psf functionality should be move into its own file.